src/Components/Modals/DeleteAdditionalRepositoriesModal.js

```diff
@@ -0,0 +1,141 @@
+import useNotifications from '../../hooks/useNotifications';
+import { useQueryClient } from 'react-query';
+import useDeleteAdditionalRepositories from '../../hooks/useDeleteAdditionalRepositories';
+
+const DeleteAdditionalRepositoriesModal = (props) => {
+  const {
+    isOpen,
+    handleModalToggle,
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete,
+    setDeletedRepositories,
+  } = props;
+  const { addSuccessNotification, addErrorNotification } = useNotifications();
+  const { mutate, isLoading } = useDeleteAdditionalRepositories();
+  const queryClient = useQueryClient();
+
+  const deleteAdditionalRepositories = (
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete
+  ) => {
+    console.log(name, repositoryNameToDelete, repositoryLabelToDelete);
+    const payload = [
+      {
+        repositoryLabel: repositoryLabelToDelete,
+        repositoryName: repositoryNameToDelete,
+      },
+    ];
+    console.log('Request Payload:', payload);
+
+    mutate(
+      { name, payload },
+      {
+        onSuccess: (data, queryName) => {
+          const updatedData = data?.filter(
+            (entry) => entry.repositoryName !== repositoryNameToDelete
+          );
+          queryClient.setQueryData(queryName, updatedData);
+          addSuccessNotification(
+            `Additional repository ${repositoryNameToDelete} deleted`
+          );
+          handleModalToggle(true);
+          setDeletedRepositories((prevState) => [
+            ...prevState,
+            {
+              repositoryName: repositoryNameToDelete,
+              repositoryLabel: repositoryLabelToDelete,
+            },
+          ]);
+        },
+        onError: () => {
+          addErrorNotification('Something went wrong. Please try again');
+          handleModalToggle();
+        },
+      }
+    );
+    mutate;
+  };
+  const actions = [
+    <Button
+      key="confirm"
+      variant="danger"
+      isLoading={isLoading}
+      onClick={() =>
+        deleteAdditionalRepositories(
+          name,
+          repositoryNameToDelete,
+          repositoryLabelToDelete
+        )
+      }
+      isDisabled={isLoading}
+      spinnerAriaValueText="Removing repository"
+    >
+      {isLoading ? 'Removing repository' : 'Remove repository'}
+    </Button>,
+    <Button
+      key="cancel"
+      variant="link"
+      onClick={handleModalToggle}
+      isDisabled={isLoading}
+    >
+      Cancel
+    </Button>,
+  ];
+
+  const title = (
+    <>
+      <TextContent>
+        <Text component={TextVariants.h2}>
+          <ExclamationTriangleIcon size="md" color="#F0AB00" />
+          {''} Remove repository?
+        </Text>
+      </TextContent>
+    </>
+  );
+
+  const content = () => (
+    <TextContent>
+      <Text component={TextVariants.p}>
+        <b>{repositoryNameToDelete}</b> will no longer be enabled when
+        registering with this activation key.
+      </Text>
+    </TextContent>
+  );
```

Comment:
This can just be a variable - it doesn't need to be a function since it returns the same thing every time